src/game/state/availablePointsForInsect.js

```diff
@@ -61,5 +61,25 @@ export const availablePointsForInsect = {
       currentPoints = subtract(newPoints, visited);
     }
     return currentPoints;
+  },
+  grasshopper: ({G, currentInsect}) => {
+    const insectsPoints = G.insects.map(({point}) => point);
+    const neighborInsectPoints = getNeighbors(currentInsect.point)
+      .filter(n =>  insectsPoints.some(i => isSame(n)(i)));
+
+    const result = [];
+
+    for (let i = 0; i < neighborInsectPoints.length; i++) {
```

Comment:
Would you prefer a `map` call here instead of the for loop?

Reply:
Yes, something like this perhaps?

```js
const result = getNeighbors(currentInsect.point)
    .filter(n => insectsPoints.some(isSame(n))
    .map(point => calculateVector(currentInsect.point, point))
    .map(jumpVector => {
        let jumpTarget = getPointByVector(currentInsect.point, jumpVector);
        while(insectsPoints.some(isSame(jumpTarget))) {
            jumpTarget = getPointByVector(jumpTarget, jumpVector);
        }
        return jumpTarget;
    });
```

or even like this if we cold partially apply ```calculateVector```:

```js
const result = getNeighbors(currentInsect.point)
    .filter(n => insectsPoints.some(isSame(n))
    .map(calculateVector(currentInsect.point))
    .map(jumpVector => {
        let jumpTarget = getPointByVector(currentInsect.point, jumpVector);
        while(insectsPoints.some(isSame(jumpTarget))) {
            jumpTarget = getPointByVector(jumpTarget, jumpVector);
        }
        return jumpTarget;
    });
```

Reply:
And if you want better naming, you could name the callbacks something sensible:

```js
const isInsect = n => insectsPoints.some(isSame(n);
const stepUntilNotOnInsect = jumpVector => {
    let jumpTarget = getPointByVector(currentInsect.point, jumpVector);
    while(insectsPoints.some(isSame(jumpTarget))) {
        jumpTarget = getPointByVector(jumpTarget, jumpVector);
    }
    return jumpTarget;
};
const result = getNeighbors(currentInsect.point)
    .filter(isInsect)
    .map(calculateVector(currentInsect.point))
    .map(stepUntilNotOnInsect);
```

I don't always do this step.